wordpress-mobile/WordPress-iOS
# Posts & Pages: Move authors filter button to navigation bar

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/21715

To test:

- Verify that the authors filters is now displayed in the navigation bar
- Verify that the filter is available for both Posts and Pages
- Verify that after a fresh install the default filter is "Everyone"

<img width="340" alt="Screenshot 2023-10-13 at 9 32 39 PM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/b23371f1-e39d-40af-acc6-2f138e56b779">


## Regression Notes
1. Potential unintended areas of impact: Posts & Pages
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
